microceph/api/types/osds.go

```diff
@@ -0,0 +1,8 @@
+// Package types provides shared types and structs.
+package types
+
+// OsdPut holds data structure for updating the state of osd service
+type OsdPut struct {
+	State    string `json:"state" yaml:"state"`
+	Location string `json:"location" yaml:"location"`
```

Comment:
I can see the Location field being updated in the client code, but on server (API) side it's not being used. 
I think we could omit the Location field in this struct and instead rely on the targetting mechanism of microcluster; by convention cli commands that support this have a `--target` argument to address a specific node.

I see below you're already running the c.UseTarget() method, that should take care of the location.